src/ModelContextProtocol/Shared/McpSession.cs

```diff
@@ -35,6 +34,8 @@ internal sealed class McpSession : IDisposable
     private readonly NotificationHandlers _notificationHandlers;
     private readonly long _sessionStartingTimestamp = Stopwatch.GetTimestamp();
 
+    private readonly DistributedContextPropagator? _propagator = DistributedContextPropagator.Current;
```

Comment:
DistributedContextPropagator.Current never returns null, right?

Reply:
user can in theory set it to null. But that'd cause a lot of issues, so let's assume it's never null.

Reply:
> user can in theory set it to null

Doesn't look like it
https://github.com/dotnet/runtime/blob/464e5fe6fbe499012445cbd6371010748b89dba3/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DistributedContextPropagator.cs#L72-L84